src/cli_util.rs

```diff
@@ -1673,10 +1672,12 @@ impl AsRef<str> for DescriptionArg {
     }
 }
 
-pub fn complete_newline(s: &mut String) {
+pub fn complete_newline(s: impl Into<String>) -> String {
```

Comment:
Couldn't you use

```rust
pub fn complete_newline(mut s: String) -> String {
  … // test + push('\n') if needed
  s
}
```

which does not create a new `String` and is chainable?

Reply:
Do you mean something like `(s: &str) -> Cow<str>` to eliminate allocation for `&str`?

I think `Into<String>` here is basically the same as `String` if `s` is `String`.

Reply:
You are right, it would have the same effect. `complete_newline()` is only called with a `String` so far, so using a `Cow` here would be overkill. Closing this conversation.